src/Features/VisualBasic/Portable/RemoveSharedFromModuleMembers/VisualBasicRemoveSharedFromModuleMembersCodeFixProvider.vb

```diff
@@ -0,0 +1,78 @@
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.RemoveSharedFromModuleMembers
+    <ExportCodeFixProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicRemoveSharedFromModuleMembersCodeFixProvider)), [Shared]>
+    Friend NotInheritable Class VisualBasicRemoveSharedFromModuleMembersCodeFixProvider
+        Inherits SyntaxEditorBasedCodeFixProvider
+
+        <ImportingConstructor>
+        <SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification:="Used in test code: https://github.com/dotnet/roslyn/issues/42814")>
+        Public Sub New()
+        End Sub
+
+        ' Methods in a Module cannot be declared '{0}'.
+        Private Const BC30433 As String = NameOf(BC30433)
+
+        ' Events in a Module cannot be declared '{0}'.
+        Private Const BC30434 As String = NameOf(BC30434)
+
+        ' Properties in a Module cannot be declared '{0}'.
+        Private Const BC30503 As String = NameOf(BC30503)
+
+        ' Variables in Modules cannot be declared '{0}'.
+        Private Const BC30593 As String = NameOf(BC30593)
+
+        Public Overrides ReadOnly Property FixableDiagnosticIds As ImmutableArray(Of String) = ImmutableArray.Create(
+            BC30433, BC30434, BC30503, BC30593)
+
+        Friend Overrides ReadOnly Property CodeFixCategory As CodeFixCategory = CodeFixCategory.Compile
+
+        Public Overrides Function RegisterCodeFixesAsync(context As CodeFixContext) As Task
+            context.RegisterCodeFix(
+                New MyCodeAction(Function(ct) FixAsync(context.Document, context.Diagnostics(0), context.CancellationToken)),
+                context.Diagnostics)
+            Return Task.CompletedTask
+        End Function
+
+        Protected Overrides Function FixAllAsync(document As Document, diagnostics As ImmutableArray(Of Diagnostic), editor As SyntaxEditor, cancellationToken As CancellationToken) As Task
+            For Each diagnostic In diagnostics
+                Dim tokenToRemove = diagnostic.Location.FindToken(cancellationToken)
+                If Not tokenToRemove.IsKind(SyntaxKind.SharedKeyword) Then
+                    Continue For
+                End If
+                Dim node = diagnostic.Location.FindNode(cancellationToken)
+                Dim newNode = GetReplacement(document, node)
+                editor.ReplaceNode(node, newNode)
+            Next
+            Return Task.CompletedTask
+        End Function
+
+        Private Shared Function GetReplacement(document As Document, node As SyntaxNode) As SyntaxNode
+            If TypeOf node Is FieldDeclarationSyntax OrElse
+               TypeOf node Is MethodBaseSyntax Then
+                Dim generator = SyntaxGenerator.GetGenerator(document)
+                Return generator.WithModifiers(node, generator.GetModifiers(node).WithIsStatic(False))
+            End If
+            Return node
```

Comment:
this implies that we might not be able fix some scenarios.  that should never happen if we've offered a fix.  so if this can hapn, thi sneeds to be checked in RegisterCodeFixesAsync

Reply:
@CyrusNajmabadi I can't figure a case when this can happen. Do I need to completely omit the "If"? Or are there any cases that I'm missing?

Reply:
either we need to prove it can't happen, in which case we can remove hte if and put a contract.throw call there instead.  or we have to assume it may happen, in which case we have to plan for it in the 'register' portion of hte fixer.

Reply:
In general you need to do the latter, as these diagnostics are produced by a black box, and could change at any point in the future (i.e. compiler might choose to report it on some new construct/location that we're not expecting.